lib/configuration-reader.js

```diff
@@ -0,0 +1,90 @@
+import path from 'path';
+import glob from 'glob';
+
+class ConfigurationReader {
+	constructor(options) {
+		const defaultOptions = {
+			path: './BuildConfiguration',
+			prefix: 'config.',
+			ext: '.js',
+			env: null
+		}
+
+		this.options = Object.assign(defaultOptions, options);
+
+		return this.getConfiguration();
+	}
+
+	getConfiguration() {
+		const options = this.options;
+
+		//const defaultConfigGlob = `${options.path}/default/*${options.ext}`;
+		//const localConfigGlob = `${options.path}/local/*${options.ext}`;
+		const configGlob = `${options.path}/{default,local,env-${options.env}}/*${options.ext}`;
+
+		//console.log('Config glob is %s', configGlob);
+
+		const configFiles = glob.sync(configGlob);
+
+		//console.log('Configfiles found: ', configFiles);
+
+		let configurations = [];
+
+		// Merge all files into an config object
+		configFiles.forEach(function(configFile) {
+
+			var match = /\/([a-zA-ZåäöÅÄÖ\s\-]+)\/Config\.?([a-zA-ZåäöÅÄÖ\s\-]*)\.?js/gi.exec(configFile);
+
+			if(!match) {
+				console.error(configFile + ' is not a valid name');
+			}
+
+			//console.log('rqp', path.relative(__dirname, configFile));
+			const configuration = require(path.relative(__dirname, configFile));
+
+			const config = {
+				env: match[1],
+				scope: match[2],
+				config: configuration
+			}
+
+			//console.log('configFiles', configFile, config);
+
+			configurations.push(config);
+			//configArrys.push(require(configFile));
+		});
+
+
+		// TODO: Sort configs to ensure correct merge order.
```

Comment:
Tests pass for default config is overwritten by environment/local config. Is this TODO obsolete? If not I'd say that this needs to be there to ensure ConfigurationReader works as intended (doesn't seem like a big thing to sort?)